slim/datasets/download_and_convert_cifar.py

```diff
@@ -0,0 +1,367 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+r"""Downloads and converts cifar10/100 data to TFRecords of TF-Example protos.
+
+This module downloads the cifar data, uncompresses it, reads the files that
+make up the cifar data and creates two TFRecord datasets: one for train and one
+for test. Each TFRecord dataset is comprised of a set of TF-Example protocol
+buffers, each of which contain a single image and label.
+
+The script should take several minutes to run.
+
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from six.moves import cPickle
+import os
+import six
+import sys
+import tarfile
+
+import numpy as np
+from six.moves import urllib
+import tensorflow as tf
+
+from datasets import dataset_utils
+
+# The URLs where the CIFAR data can be downloaded.
+_DATA_URL = {
+  'cifar10': 'https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz',
+  'cifar100': 'https://www.cs.toronto.edu/~kriz/cifar-100-python.tar.gz'
+}
+
+_DATA_DIR = {
+  'cifar10': 'cifar-10-batches-py',
+  'cifar100': 'cifar-100-python',
+}
+
+# The number of training files.
+_NUM_TRAIN_FILES = {
+  'cifar10': 5,
+  'cifar100': 1,
+}
+
+# The height and width of each image.
+_IMAGE_SIZE = 32
+
+# The names of the classes.
+_CLASS_NAMES = {
```

Comment:
There class names for CIFAR10 were included in the python file, so I did the same for CIFAR100.  Alternatively, they could be loaded from the `meta` file.